crates/wasi-provider/src/states/finished.rs

```diff
@@ -0,0 +1,10 @@
+use crate::PodState;
+use kubelet::state::prelude::*;
+
+state!(
+    /// Pod execution completed with no errors.
+    Finished,
```

Comment:
The difference between `Finished` and `Completed` is elusive, especially as the documentation comment for `Finished` uses the term `completed`!  Can we find names that better express the distinction?  (I understand that the answer may be "no.")

Reply:
oh! I see the difference between the two states now. Yes, this should be Completed, and the current "Completed" state is actually the Terminated state.

Reply:
renamed to reflect this.